src/boot/efi/efi-string.c

```diff
@@ -42,3 +43,67 @@ size_t strlen8(const char *s) {
 size_t strlen16(const char16_t *s) {
         return strnlen16(s, SIZE_MAX);
 }
+
+#define TOLOWER(c) ((c >= 'A' && c <= 'Z') ? c + ('a' - 'A') : c)
+
+char tolower8(char c) {
+        return TOLOWER(c);
+}
+
+char16_t tolower16(char16_t c) {
+        return TOLOWER(c);
+}
+
+#define STRNCASECMP_U(cmp_type, case)                \
+        ({                                           \
+                if (!s1 || !s2)                      \
+                        return CMP(s1, s2);          \
+                                                     \
+                while (n > 0) {                      \
+                        cmp_type c1 = *s1, c2 = *s2; \
+                        if (case) {                  \
```

Comment:
I prefer `case` -> `tolower` or something, as "case" is reserved. Of course, this is a macro. But still...